### PR TITLE
FORMS-1207 - New Map Component for CHEFS

### PR DESCRIPTION
**Pull request by abhilash-aot**
# Description

Map component with Point and Circle Marker for getting the latitude and longitude.
Parent ticket : https://bcdevex.atlassian.net/browse/FORMS-1207

This PR introduces a new map component in the Chefs application, integrated into the form designer. This component allows designers to set the center point, default markers, and enables users to view readonly maps within forms. The map component supports  point ,line, polygon and circle markers to capture latitude and longitude coordinates.

Further documentation on map can be found here  at techdocs :  https://github.com/RyanBirtch-aot/common-hosted-form-service-techdocs/blob/docs/map-init/docs/Components/Form-Builder/BC-Government.md#map-component 

You can test the Map component in the PR environment : https://chefs-dev.apps.silver.devops.gov.bc.ca/pr-1437

The techdoc PR: https://github.com/bcgov/common-hosted-form-service-techdocs/pull/54

This change is required to enhance the form designer capabilities, allowing users to interact with and visualize geographical data directly within forms. It solves the problem of needing an intuitive and interactive way to input and display location data.

## Types of changes

- [x] feat (a new feature)

## Checklist

- [x] I have read the [CONTRIBUTING](/bcgov/common-hosted-form-service/blob/main/CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have run the npm script lint on the frontend and backend
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] I have approval from the product owner for the contribution in this pull request

## Further comments

This addition aims to provide a robust and user-friendly way to manage and display geospatial data in forms. Alternatives considered included integrating with other map services or using simpler text input fields for coordinates, but those did not meet the usability requirements.

### CODE REVIEW

File: components/src/components/Map/Component.ts [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+  mapService: MapService;
+
+  constructor(component, options, data) {
+    super(component, options, data);
+    this.componentID = super.elementInfo().component.id;
+  }
+
+  render() {
+    return super.render(
+      `<div id="map-${this.componentID}" style="height:400px; z-index:1;"> </div>`
```
Comment (usingtechnology) [inline components/src/components/Map/Component.ts:45]: can we make the height a property the designer sets?

File: components/src/components/Map/editForm/Component.edit.data.ts [hunk 2]
```diff
@@ -0,0 +1,122 @@
+export default {
+  key: 'customData',
+  label: 'Data',
+  weight: 20,
+  components: [
+    {
+      html: '<h2>Default Values</h2>',
+      key: 'simplecontent1',
+      type: 'simplecontent',
```
Comment (usingtechnology) [inline components/src/components/Map/editForm/Component.edit.data.ts:9]: can we use the formio version not our custom simple ones? at some point we may remove the simple ones and then we'd need to know to update this component too.

**Review by usingtechnology — CHANGES_REQUESTED** [review]
Screenshot 2024-11-01 at 12 51 23 PM
Screenshot 2024-11-01 at 12 50 50 PM
Looks good, a couple of questions for you.
Also, the tooltip or whatever on the cursor gets hidden by the popout menus (like finish, cancel on draw) and it's confusing (to me). perhaps we can get rid of the notes on the cursor?

File: components/src/components/Map/Component.ts (hunk 1 above)
Comment (RyanBirtch-aot) [reply components/src/components/Map/Component.ts:45]: Yes, but height customizability never really came up in the ux reviews. I can pass it by minshen

File: components/src/components/Map/editForm/Component.edit.data.ts (hunk 2 above)
Comment (RyanBirtch-aot) [reply components/src/components/Map/editForm/Component.edit.data.ts:9]: updated in latest commit

Comment (usingtechnology) [reply components/src/components/Map/editForm/Component.edit.data.ts:9]: missed the simplecontent in Component.edit.data.ts

File: components/src/components/Map/Component.ts (hunk 1 above)
Comment (RyanBirtch-aot) [reply components/src/components/Map/Component.ts:45]: After talking to minshen, it doesn't seem like a high priority for designers to choose this. It's possible we can take a look at an advanced tab with some of these settings down the road.

File: components/src/components/Map/editForm/Component.edit.data.ts (hunk 2 above)
Comment (RyanBirtch-aot) [reply components/src/components/Map/editForm/Component.edit.data.ts]: Thanks for catching that!

**Review by usingtechnology — APPROVED** [review]

**Pull request merged**